mcs/class/System.Private.CoreLib/System.Threading/Mutex.cs

```diff
@@ -1,14 +1,80 @@
+using System.Runtime.CompilerServices;
+using System.IO;
+
 namespace System.Threading
 {
 	partial class Mutex
 	{
-		void CreateMutexCore (bool initiallyOwned, string name, out bool createdNew) => throw new NotImplementedException ();
-
-		static OpenExistingResult OpenExistingWorker (string name, out Mutex result) => throw new NotImplementedException ();
+		Mutex (IntPtr handle) => Handle = handle;
 
 		public void ReleaseMutex ()
 		{
-			throw new NotImplementedException ();
+			if (!ReleaseMutex_internal (Handle))
+				throw new ApplicationException ("Mutex is not owned");
```

Comment:
This is an unlikely right exception, what does core throw?

Reply:
Surprisingly enough that's what CoreRT throws on Unix: `throw new ApplicationException(SR.Arg_SynchronizationLockException);`

Reply:
...and CoreCLR as well: https://github.com/dotnet/coreclr/blob/cf85f56e473b6cdc4a68e9758f57dc571e14c9cb/src/System.Private.CoreLib/shared/System/Threading/Mutex.Windows.cs#L95

Reply:
@stephentoub could you please confirm this^ is intentional

Reply:
It's intentional, unfortunately.  It's what netfx has used forever:
https://referencesource.microsoft.com/#mscorlib/system/threading/mutex.cs,425
and the cost/benefit to changing it at this point is not favorable.